### DISCUSSION TITLE
How to use custom module aliases?

### DISCUSSION THREAD

**Discussion by ttbarnes** [cypress-io/cypress#25309] [Questions and Help] [answered]
Hiya, I have a monorepo and one of the directories is for cypress tests. In other servies we can simply add `_moduleAliases` to package.json and we can then import things from a sibling directory, e.g:

```js
// package.json 
{
  "name": "The project",
  "_moduleAliases": {
    "@constants": "../constants"
  }
}

// ui/some/file.js
import CONSTANTS from '@constants`;
```

I've tried doing the exact same for the package.json in the cypress tests directory, but it can't resolve the module when running the tests.

Is it possibile to use custom module aliases in cypress? Can't see any documentation about this and various other posts/discussions seem to go down a rabbit hole and not up to date.

**Comment by kryshtopus** [comment]
You can use https://www.npmjs.com/package/module-alias with updated tsconfig.json @ttbarnes

**Comment by jennifer-shehane** [comment, marked as answer]
Cypress doesn’t read package.json `_moduleAliases` or tsconfig `compilerOptions.paths` when bundling spec/support files, those aren’t bundler features. You need to tell the bundler (the webpack preprocessor, by default) about the aliases:

- Add a `resolve.alias` entry to the webpack config you pass to the preprocessor, or
- Reuse your tsconfig paths by adding tsconfig-paths-webpack-plugin to `resolve.plugins`.

New docs here: https://on.cypress.io/preprocessors-api#Resolving-import-path-aliases
